### ISSUE TITLE
Lqip cannot process image name with a number at the end

### ISSUE THREAD

**Issue by sira313** [hexojs/hexo-filter-lqip#47]
my file name  is `/img/asset/Aronath2.jpg` 
here's the `--debug` result
```
03:37:01.192 ERROR Can not find file: /img/asset/Aronath2jpg
Error: Can not find file: /img/asset/Aronath2jpg
    at /home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/hexo-filter-lqip/lqip/process.js:29:29                                                                                          
    at assignReplacement (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/string-replace-async/index.js:34:18)                                                                         
    at /home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/string-replace-async/index.js:53:10                                                                                             
    at Array.map (<anonymous>)                                                                                                                                                                       
    at concurrency (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/string-replace-async/index.js:52:25)                                                                               
    at processString (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/string-replace-async/index.js:71:9)                                                                              
    at fn (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/string-replace-async/index.js:81:26)                                                                                        
    at stringReplaceAsync (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/string-replace-async/index.js:88:9)                                                                         
    at processType (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/hexo-filter-lqip/lqip/process.js:23:10)                                                                            
    at /home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/hexo-filter-lqip/lqip/process.js:59:12                                                                                          
    at tryCatcher (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/util.js:16:23)                                                                                  
    at Object.gotValue (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/reduce.js:157:18)                                                                          
    at Object.gotAccum (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/reduce.js:144:25)                                                                          
    at Object.tryCatcher (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/util.js:16:23)                                                                           
    at Promise._settlePromiseFromHandler (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/promise.js:512:31)                                                       
    at Promise._settlePromise (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/promise.js:569:18)                                                                  
    at Promise._settlePromiseCtx (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/promise.js:606:10)                                                               
    at _drainQueueStep (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/async.js:142:12)                                                                           
    at _drainQueue (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/async.js:131:9)                                                                                
    at Async._drainQueues (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/async.js:147:5)                                                                         
    at Immediate.Async.drainQueues (/home/sira/Templates/Aflasio.com/www.aflasio.com/node_modules/bluebird/js/release/async.js:17:14)                                                                
    at runCallback (timers.js:794:20)                                                                                                                                                                
    at tryOnImmediate (timers.js:752:5)                                                                                                                                                              
    at processImmediate [as _immediateCallback] (timers.js:729:5)
```

> if possible, please add feature to make it simple just add class name to use lqip in the future
for example: 
```html
<img class="lqip" src="/myimage.jpg"/>
```
your plugins is awesome
thanks

**Comment by ertrzyiks** [comment]
Hey @sira313, thanks for taking time and reporting this issue.

Since I'm not able to reproduce this issue by just adding a number to the file name, I would need more details on how do you use the plugin.

Do you have a public repo where I can reproduce this issue?
I would check what is passed as a `path` here: https://github.com/hexojs/hexo-filter-lqip/blob/d07f7abe6238775a856b68602bd8a3135cc732df/lqip/index.js#L65 Is there any chance that you pass a file name without that dot to the `lqipFor` helper?

**Comment by sira313** [comment]
The only issue i've found is just like i told before.
Here's my repo that i use hexo-filter-lqip
https://gitlab.com/sira313/www.aflasio.com/commit/8a630d457ce21cdf5d938e436da0b4e1a3c5a550

**Comment by ertrzyiks** [comment]
@sira313 thanks for sharing the link. I was able to reproduce the issue you described on my computer and noticed that the problem is occuring only if you have `hexo-neat` plugin enabled. That plugin transforms meta data created by lqip plugin.

I'm checking if we can do anything about it to prevent a clash between those plugins.

**Comment by ertrzyiks** [comment]
@sira313 I submitted a PR which should prevent such issues from happening. https://github.com/hexojs/hexo-filter-lqip/pull/50

I will fix specs and test it a bit before releasing a new version of this plugin. Meanwhile, you can check it out to see if it solves your issue, for example by installing it from the branch with

npm
```
npm install hexojs/hexo-filter-lqip#encode-path-in-content
```

or yarn
```
yarn add hexojs/hexo-filter-lqip#encode-path-in-content
```

**Comment by ertrzyiks** [comment]
I actually merged the fix and released 1.3.1 with a patch for this issue. Please let me know if it's solved for you.

**Comment by sira313** [comment]
> I actually merged the fix and released 1.3.1 with a patch for this issue. Please let me know if it's solved for you.

Thanks, it's solved. My only problem is just i don't know how write that in my custom theme [griddy](https://github.com/sira313/hexo-theme-griddy) yet. No problem, it's just 'cause i am not a coder 😊

**Comment by ertrzyiks** [comment]
@sira313 let's open another issue for that #51. The problem is it's impossible to provide a solution that works for everyone and all the cases. 

Meanwhile, you can see how is it done in the example https://github.com/ertrzyiks/hexo-lqip-example and copy-paste some code from there.
